src/components/TodoLayout/CreateTodoModal/CreateTodoModal.tsx

```diff
@@ -0,0 +1,149 @@
+import { FC, FormEvent, useState } from "react";
+
+import { Button } from "@/components/ui/Button/Button.tsx";
+import { Input } from "@/components/ui/Input/Input.tsx";
+import { Modal } from "@/components/ui/Modal/Modal.tsx";
+
+import { useAppDispatch } from "@/hooks/useAppDispatch.ts";
+import { useAppSelector } from "@/hooks/useAppSelector.ts";
+import { useModalState } from "@/hooks/useModalState.ts";
+import { addTodo, setTodoTitle } from "@/store/actions/todoActionCreators.ts";
+
+import { checkOnValidField } from "@/utils/checkOnValidField.ts";
+import { setMaxTimeToDate, setMinTimeToDate, setMinutesToDate } from "@/utils/setTimeToDate.ts";
+
+import cn from "classnames";
+
+import { TodoScheme } from "@/common/schemes/TodoScheme.ts";
+
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import { format } from "date-fns";
+
+import { Formik } from "formik";
+
+import styles from "./CreateTodoModal.module.scss";
+import moduleStyles from "@/styles/ModalCommom.module.scss";
+
+export const CreateTodoModal: FC = () => {
+  const [modalActive, setModalActive] = useModalState("createTodoModal");
+
+  const [expirationDate, setExpirationDate] = useState<Date | null>(null);
+
+  const { title } = useAppSelector(state => state.todoReducer);
+  const dispatch = useAppDispatch();
+
+  const closeModal = (): void => {
+    setModalActive(false);
+    dispatch(setTodoTitle({ title: "" }));
+  };
+
+  const handleChangeTodoTitle = (
+    e: FormEvent<HTMLInputElement>,
+    setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void,
+  ): void => {
+    const newValue = checkOnValidField(e.currentTarget.value);
+    setFieldValue("title", newValue);
+    dispatch(setTodoTitle({ title: newValue }));
+  };
+
+  const handleDateChange = (date: Date | null, setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void): void => {
+    if (date) {
+      if (new Date() > date) {
+        setExpirationDate(setMinutesToDate(new Date(), 5));
+        setFieldValue("expirationDate", setMinutesToDate(new Date(), 5));
+        return;
+      }
+
+      setExpirationDate(date);
+      setFieldValue("expirationDate", date);
+    }
+  };
+
+  const handleSubmit = () => {
+    setModalActive(false);
+    if (expirationDate) {
+      dispatch(
+        addTodo({
+          createdDate: format(new Date(), "dd.MM.yyyy HH:mm"),
```

Comment:
Again - you should have the same format within the app - so, it's better to have const